embedded/ee10-servlet-server/src/main/java/examples/EmbedMe.java

```diff
@@ -51,6 +49,7 @@ public static Server newServer(int port)
         context.setBaseResource(baseResource);
         context.setContextPath("/");
         context.setWelcomeFiles(new String[]{"index.html", "welcome.html"});
+        context.setAttribute("org.eclipse.jetty.server.webapp.ContainerIncludeJarPattern", ".*/");
```

Comment:
Set the pattern properly, don't include 100% of the container.
As on Maven, this even includes all of the maven deps, and the maven plugins too.

In fact, the default pattern in ee10 should work out of the box.

Reply:
What should be the pattern? When I removed this line (using default), test was failing. I changed to `.*/classes/*` now.

Reply:
There are two patterns that impact how the webapp scanning operates.

1. The Container level (`org.eclipse.jetty.server.webapp.ContainerIncludeJarPattern`) which scans for things that are not part of the webapp.
     By default, it will not scan the container, unless this pattern is defined.
2. The WebApp level (`org.eclipse.jetty.server.webapp.WebInfIncludeJarPattern`) which scans for things that belong specifically to the webapp.
     By default, this will scan all artifacts on the webapp (like `WEB-INF/lib/` and `classes/`) unless the pattern is defined.

For the container, you might need to support servlet, and jsp, and taglib.

For servlet only ...

``` java
webappContext.setAttribute(
    "org.eclipse.jetty.server.webapp.ContainerIncludeJarPattern",
    ".*/jakarta.servlet-api-[^/]*\\.jar$");
```

For servlet, with JSP, and even Taglib support ...

``` java
webappContext.setAttribute(
    "org.eclipse.jetty.server.webapp.ContainerIncludeJarPattern",
    ".*/jakarta.servlet-api-[^/]*\\.jar$|.*/jakarta.servlet.jsp.jstl-.*\\.jar$|.*/[^/]*taglibs.*\\.jar$");
```

Add more jar patterns for other jars that you might need from the container (which shouldn't be much).
Point is, don't open everything on the container for scanning, that's just wrong way to use the pattern, wrong for performance as you scan everything, wrong for security as you never want to open everything to the webapp, etc ...

Reply:
The overwhelming majority of embedded use cases never scan, as that's an anti-pattern for embedded use cases (which focus on absolute control, and startup performance).

They instead define the webapp configuration entirely in code for quick startup.